### DISCUSSION TITLE
[help] When is it best to create an R package out of the R folder?

### DISCUSSION THREAD

**Discussion by asadow** [ropensci/targets#1105] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi, my targets project is also an R package. I'm curious whether I should instead separate out the R package of functions and tests and use package-based invalidation in my targets project as per [this chapter](https://books.ropensci.org/targets/packages.html#package-based-invalidation).

Do you have any feedback in this regard?

**Comment by wlandau** [comment, marked as answer]
From experience, I think ideal `targets` pipeline is pretty much a research compendium (https://github.com/ropensci/rrrpkg) but not quite a package. If you are developing a new method, for example, I recommend packaging up the functions and unit tests of the method, but have a separate non-package file system for a pipeline that applies the method to a specific data analysis project.

**Reply by asadow** [reply]
Great. Thank you, Will, for your insight and the link.
